modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -1302,6 +1336,32 @@ definition {
 		AppBuilderAdmin.validateIfSearchItemIsNotFound(itemName = "Apple");
 	}
 
+	@description = "Unpublish and Republish an App"
+	@priority = "5"
+	test UnpublishAndRepublish {
+		property portal.acceptance = "false";
+
+		AppBuilderAdmin.addCustomObject(
+			customObjectName = "my custom object",
+			isCreatingFormView = "true",
+			isCreatingViaPlusIcon = "true");
+
+		AppBuilderAdmin.createFormViewAndTableView(
+			fieldType = "Text Field",
+			formViewName = "Form View Title",
+			tableViewName = "Table View Title");
+
+		App.add(
+			appTitle = "Test App",
+			deployOption = "Product Menu",
+			formViewName = "Form View Title",
+			tableViewName = "Table View Title");
+
+		AppBuilderAdmin.undeployApp(itemName = "Test App");
+
+		AppBuilderAdmin.deployApp(itemName = "Test App");
```

Comment:
Hi @nkkwon ,
Checking the purpose of the tests, I believe we should add a few more validations:

_1) Undeploy the app and make sure undeployed App must become inaccessible for users in all deployment methods configured_

_2) Redeploy the app and make sure undeployed App must become accessible again for users in all deployment methods configured_

As far as I understand of the App Builder, an app can be deployed as Product Menu, Widget and Standalone. To cover the tests, I believe that more steps should be taken and more validations should be added.

1) Deploy the app as Product Menu, Widget, and Standalone and guarantee and all of these options are accessible
2) Undeploy the app and then guarantee that app can't be accessed by Product Menu, Widget, and Standalone
3) Redeploy and validate all 3 options once again

Please correct if I am missing something, but checking the macros you created and the validations performed, I am not seeing these validations.

Does it make sense? Please let me know your thoughts.

FYI @alexcandidos